Library/Homebrew/cmd/update.sh

```diff
@@ -556,6 +556,9 @@ EOS
     [[ -d "${DIR}/.git" ]] || continue
     cd "${DIR}" || continue
 
+    # Git's fsmonitor prevents the release of our locks
+    git config --bool core.fsmonitor false
```

Comment:
Will this work for `fsmonitor` processes that are outside of Homebrew's repo? If you enabled the monitor in a separate directory? This might be overkill (pun intended), but maybe `update` can kill the `fsmonitor` daemon during an update. The better solution seems like introspecting what causes this message 'Another active Homebrew update process is already in progress' which seems to be caused by (without deeply looking at `brew`'s source, finding `homebrew` in the open processes path). 

```sh
λ ps | grep brew
93281 ttys002    0:01.07 /opt/homebrew/Cellar/git/2.37.0/libexec/git-core/git fsmonitor--daemon run --detach --ipc-threads=8
11139 ttys005    0:01.32 /opt/homebrew/Cellar/git/2.37.0/libexec/git-core/git fsmonitor--daemon run --detach --ipc-threads=8
89381 ttys007    0:00.30 /opt/homebrew/Cellar/git/2.37.0/libexec/git-core/git fsmonitor--daemon run --detach --ipc-threads=8
28329 ttys008    0:00.24 /opt/homebrew/Cellar/git/2.37.0/libexec/git-core/git fsmonitor--daemon run --detach --ipc-threads=8
16062 ttys011    0:00.50 /opt/homebrew/Cellar/git/2.37.0/libexec/git-core/git fsmonitor--daemon run --detach --ipc-threads=8
23004 ttys012    0:00.02 /opt/homebrew/Cellar/git/2.37.0/libexec/git-core/git fsmonitor--daemon run --detach --ipc-threads=8
23007 ttys012    0:00.01 /opt/homebrew/Cellar/git/2.37.0/libexec/git-core/git fsmonitor--daemon run --detach --ipc-threads=8
23013 ttys012    0:00.02 /opt/homebrew/Cellar/git/2.37.0/libexec/git-core/git fsmonitor--daemon run --detach --ipc-threads=8
24442 ttys012    0:00.00 grep brew
```

Reply:
> Will this work for `fsmonitor` processes that are outside of Homebrew's repo? If you enabled the monitor in a separate directory?

Do these actually cause issues? I assumed the issue was caused by the initial spawning of the daemon inheriting the file descriptors from `update.sh` which wouldn't happen if it was spawned from another process, but I never managed to test if that was the case. If this theory is correct, this is in part our issue since we should ideally have close-on-exec behaviour for lock descriptors but don't.

If it's instead the case that the daemon just holds on to any file no matter what, then that feels more like a general Git bug since that behaviour sounds very problematic and uncontrollable.

Reply:
Whether the `fsmonitor` is inside a homebrew git directory or another user owned directory shouldn't be a concern of doing a `git pull` on the directory. It's only monitoring the files there. I also don't think we should need to kill the monitor or have the lock be concerned with it. It might just be a matter of more carefully choosing names for the lockfiles.

Reply:
I have the monitor setup globally, though it does feel invasive to have `brew` change the config there. I don't suppose it matters for brew managed git repos though, so not terribly concerned.

```
λ git config --global core.fsmonitor
true
```